components/meter/meter-linear.js

```diff
@@ -32,23 +49,61 @@ class MeterLinear extends LitElement  {
 				position: absolute;
 				top: 0;
 			}
+
+			.d2l-meter-linear-text {
+				color: var(--d2l-color-ferrite);
+				display: flex;
+				flex-direction: row;
+				justify-content:space-between;
+				margin: 0 6px;
+			}
+
+			.d2l-meter-linear-secondary {
+				align-self: flex-end;
+			}
+
 		`];
 	}
 
 	constructor() {
 		super();
-		this.value = 0;
 		this.max = 100;
+		this.percent = false;
+		this.singleLine = false;
+		this.value = 0;
 	}
 
 	render() {
 		const progressBarWidth = this.value / this.max * 100;
+		const primary = this.percent ? `${Math.floor(this.value / this.max * 100)}%` : `${this.value}/${this.max}`;
+		const secondaryText = this._formatContext(this.context, this.value, this.max);
+		const secondaryTextElement = secondaryText ? html`<div class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
+		const classSingleLine = this.singleLine ? 'd2l-meter-linear-flex' : 'd2l-meter-linear-block';
 		return html `
-			<div class="d2l-meter-linear-full-bar">
-				<div class="d2l-meter-linear-inner-bar" style="width:${progressBarWidth}%;"></div>
+			<div class="${classSingleLine}" role="img" aria-label="${secondaryText}: ${primary}; progress indicator">
+				<div class="d2l-meter-linear-full-bar">
+					<div class="d2l-meter-linear-inner-bar" style="width:${progressBarWidth}%;"></div>
+				</div>
+				<div class="d2l-body-small d2l-meter-linear-text">
+					<!-- I am inserting this space here for visual as well copy and pasting primary and secondary text gives the correct results -->
+					<div class="d2l-meter-linear-primary">${primary}&nbsp;</div>
```

Comment:
I put this space here because I wanted a natural space when these are placed on the same line. Happens when user uses single-line attribute.
This also allows a user to copy and past this element such that there is actually a space between primary and context text. Which is what you would expect when doing a copy paste. If there is a better way let me know :)

Reply:
Does the HTML comment get stripped?

Reply:
I don't know to be honest. I could remove it.

Reply:
If it doesn't get stripped, I'd remove it.  :)